sertalpbilal/FPL-Optimization-Tools
# Updates to the Docker Config and other misc fixes

This PR proposes the following changes:

* A few small tweaks to add `.gitignore` and `.gitkeep` files to the `results` folders, to ensure all results files are not added to Git.
* A few tweaks to the Docker config and setup:
     - Using a `docker-compose` file to more easily sync changes (e.g updating config) between the host and the Docker container, also simplifies the commands needed to use the Docker setup.
     - Simplification of the logic so that it copies the current local repo across to the container, rather than pulling it from Git, and keeps it in sync.

Testing Instructions

* Run `docker-compose up` in the root directory of the project. This should bring the docker container up and run the solver. (if it is run for a second time, it won't need to bring the container up again so it should just run the solver.)
* Changes made to the configuration files (e.g. `regular_settings.json`) should sync across to the Docker container, so you should be able to run the solver, make some changes to the config, then run the solver again and it will take the change into account.
